pkg/console/operator/health.go

```diff
@@ -0,0 +1,85 @@
+package operator
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"os"
+	"time"
+
+	deploymentv1 "k8s.io/api/apps/v1"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	routev1 "github.com/openshift/api/route/v1"
+
+	"github.com/openshift/console-operator/pkg/console/status"
+	deploymentsub "github.com/openshift/console-operator/pkg/console/subresource/deployment"
+	routesub "github.com/openshift/console-operator/pkg/console/subresource/route"
+)
+
```

Comment:
Pulling these out into separate funcs so we can call them at other points in the sync loop.  For example, we can check route health much earlier.

Reply:
> Pulling these out into separate funcs so we can call them at other points in the sync loop

wondering if that will be true even after we refact our big sync loop into controllers ?

Reply:
Prob will not matter at that point.  But I think we might want this now (and maybe backport).